lib/misc.c

```diff
@@ -311,3 +311,23 @@ __normalize_key(const gchar* buffer)
 {
   return str_replace_char(buffer, '-', '_');
 }
+
+gboolean
+pwrite_strict(gint fd, const void *buf, size_t count, off_t offset)
+{
+  ssize_t written = pwrite(fd, buf, count, offset);
+  gboolean result = TRUE;
+  if (written != count)
+    {
+      if (written != -1)
+        {
+          msg_error("Short written",
+                    evt_tag_int("required", count),
+                    evt_tag_int("written", written),
+                    NULL);
+          errno = ENOSPC;
```

Comment:
I think this error doesn't give enough context to understand the error. I doubt it could be much better without turning it into persist specific.

Why don't we do that? I mean I am sceptic about its reusability from other call sites, so we should move this to the persist code and fix the message there.